src/core/Config.cpp

```diff
@@ -60,39 +60,43 @@ Config::Config(const QString& fileName, QObject* parent)
 Config::Config(QObject* parent)
     : QObject(parent)
 {
-    QString userPath;
-    QString homePath = QDir::homePath();
-
-#if defined(Q_OS_UNIX) && !defined(Q_OS_MAC)
-    // we can't use QStandardPaths on X11 as it uses XDG_DATA_HOME instead of XDG_CONFIG_HOME
-    QByteArray env = qgetenv("XDG_CONFIG_HOME");
-    if (env.isEmpty()) {
-        userPath = homePath;
-        userPath += "/.config";
+    // Check if portable config is present. If not, find it in user's directory
+    QString portablePath = QDir::currentPath() + "/keepassxc.ini";
```

Comment:
@droidmonkey doesn't `QDir::currentPath` return the path from which the executable was launched, and not necessarily the path of the executable itself? Using `QCoreApplication::applicationDirPath()` would return the path of the executable instead.